### PR TITLE
fix: pseudo-transactions for `ledger` and `tx`

### PR DESCRIPTION
**Pull request by pdp2121**
## High Level Overview of Change

Currently the explicit responses for `tx` and `ledger` methods do not include psedo-transactions (EnableAmendment, UNLModify, SetFee) even though these transactions should still be found with the above methods. This PR will add these types to the responses.

The response type for `ledger` will allow specifying a custom ledger entry type.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### Did you update HISTORY.md?

- [x] Yes
- [ ] No, this change does not impact library users

### CODE REVIEW

File: packages/xrpl/src/models/methods/tx.ts [hunk 1]
```diff
@@ -39,8 +40,9 @@ export interface TxRequest extends BaseRequest {
  *
  * @category Responses
  */
-export interface TxResponse<T extends BaseTransaction = Transaction>
-  extends BaseResponse {
+export interface TxResponse<
+  T extends BaseTransaction | PseudoTransaction = Transaction,
```
Comment (ckniffen) [inline packages/xrpl/src/models/methods/tx.ts:44]: Dont all `PseudoTransaction` options extend BaseTransactions.

Would this work? 

```
T extends BaseTransaction = Transaction | PseudoTransaction,
```

Comment (pdp2121) [reply packages/xrpl/src/models/methods/tx.ts:44]: I tried that earlier but `UNLModify` does not extend `BaseTransaction`

Comment (ckniffen) [reply packages/xrpl/src/models/methods/tx.ts:44]: It should be able to.

**Comment by ckniffen** [comment]
You should also update `LedgerEntryResponse` to be a generic that defaults to `LedgerEntry` this will allow people using the library in side chains/test networks to specify ledger object types that are not officially supported yet.

https://github.com/XRPLF/xrpl.js/blob/main/packages/xrpl/src/models/methods/ledgerEntry.ts#L187C18-L202

File: packages/xrpl/src/models/ledger/Ledger.ts [hunk 2]
```diff
@@ -61,5 +62,7 @@ export default interface Ledger {
    * either JSON or binary depending on whether the request specified binary
    * as true.
    */
-  transactions?: Array<Transaction & { metaData?: TransactionMetadata }>
+  transactions?: Array<
+    (Transaction | PseudoTransaction) & { metaData?: TransactionMetadata }
```
Comment (khancode) [inline packages/xrpl/src/models/ledger/Ledger.ts:66]:
```suggestion
    (Transaction | PseudoTransaction) & { metaData?: TransactionMetadata }
```
Is it possible to combine `Transaction | PseudoTransaction` into one type? I see typing also occurs `hashTxTree`

**Review by intelliot — APPROVED** [review]
I don't have a way to test this, but LGTM

**Review by khancode — APPROVED** [review]
LGTM. Would like to have this refactored in a separate PR with transaction type `Transaction = SubmittableTransaction | PseudoTransaction`

**Pull request merged**